lfs/client.go

```diff
@@ -125,8 +126,15 @@ func Download(oid string) (io.ReadCloser, int64, error) {
 		return nil, 0, err
 	}
 	LogTransfer("lfs.data.download", res)
+	
+	buf, _ := ioutil.ReadAll(res.Body)
+	body := myReader{bytes.NewBuffer(buf)}	
```

Comment:
@rubyist just mentioned that my previous idea of setting the max idle hosts to 1 would not work at all. It'd more likely just force each request to create a new connection.

It seems weird that NTLM would require the exact same TCP connection. Is there some token or state value that we can store on the global `Config` object, perhaps? I'm starting to think I'll have to learn how NTLM works before being comfortable merging this...